cartographer_ros/cartographer_ros/node.cc

```diff
@@ -459,6 +457,55 @@ void Node::FinishAllTrajectories() {
   }
 }
 
+void Node::FinishTrajectory(const int trajectory_id) {
+  carto::common::MutexLocker lock(&mutex_);
+  map_builder_bridge_.FinishTrajectory(trajectory_id);
```

Comment:
`CHECK` that `is_active_trajectory_` is `true` before calling `FinishTrajectory()` and set to `false` afterwards?